### PR TITLE
[refactor]: bump dependencies

### PR DESCRIPTION
**Pull request by mversic**
### Linked issue

Closes #{issue_number} 

### Checklist

- [ ] I've read `CONTRIBUTING.md`
- [ ] I've used the standard signed-off commit format (or will squash just before merging)
- [ ] All applicable CI checks pass (or I promised to make them pass later)
- [ ] (optional) I've written unit tests for the code changes
- [ ] I replied to all comments after code review, marking all implemented changes with thumbs up

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -58,238 +58,301 @@ iroha_trigger_derive = { version = "=2.0.0-pre-rc.19", path = "smart_contract/tr
...
-# rust.unreachable_pub = "deny"
-# rust.unsafe_op_in_unsafe_fn = "deny"
+# lower the priority to allow overriding later
+clippy.unused_rounding = "deny"
 
-# These are up to personal taste. We don't want these to be enabled ever.
+# lower the priority to allow overriding later
 clippy.string_add = "allow"
+
+# lower the priority to allow overriding later
```
Comment (Arjentix) [inline Cargo.toml:222]: Why did you leave this comment for every lint? I think it was on a right place before

Comment (Arjentix) [reply Cargo.toml:222]: Maybe some accident automation?

File: client/tests/integration/triggers/by_call_trigger.rs [hunk 2]
```diff
@@ -337,7 +337,8 @@ fn trigger_in_genesis_using_base64() -> Result<()> {
 
     info!("WASM size is {} bytes", wasm.len());
 
-    let wasm_base64 = serde_json::json!(base64::encode(&wasm)).to_string();
+    let engine = base64::engine::general_purpose::STANDARD;
+    let wasm_base64 = serde_json::json!(base64::engine::Engine::encode(&engine, wasm)).to_string();
```
Comment (Arjentix) [inline client/tests/integration/triggers/by_call_trigger.rs:341]: Oh, why does simple `base64` encoding it should be that awkward now :(

Just complaining about API design of the crate, no objections agains this PR

**Review by Erigara — COMMENTED** [review]
LGTM, answer Daniil comment for lints and ping for approval

File: client/tests/integration/triggers/by_call_trigger.rs (hunk 2 above)
Comment (mversic) [reply client/tests/integration/triggers/by_call_trigger.rs:341]: yeah, idk. I'm not base64 expert :)

File: Cargo.toml (hunk 1 above)
Comment (mversic) [reply Cargo.toml:222]: interesting, no idea what happened

File: client/tests/integration/triggers/by_call_trigger.rs (hunk 2 above)
Comment (DCNick3) [reply client/tests/integration/triggers/by_call_trigger.rs:341]: It's a weird thing that happened recently.. There are some recommendations with a bit more normal API https://twitter.com/mycoliza/status/1708933310548935148

Comment (Arjentix) [reply client/tests/integration/triggers/by_call_trigger.rs:339]: Time to have more posts from *tech* twitter in my feed

File: client/benches/tps/dev.rs [hunk 3]
```diff
@@ -2,11 +2,6 @@
 //! using [criterion](https://github.com/bheisler/criterion.rs)
 //! for performance check during development
 #![allow(missing_docs)]
-#![allow(
-    clippy::arithmetic_side_effects,
-    clippy::std_instead_of_core,
-    clippy::std_instead_of_alloc
-)]
```
Comment (Arjentix) [inline client/benches/tps/dev.rs:9]: So you just removed these lints from Cargo.toml?

**Review by Arjentix — APPROVED** [review]

File: client/benches/tps/dev.rs (hunk 3 above)
Comment (mversic) [reply client/benches/tps/dev.rs:9]: it's not me who removed them. They have been allowed long ago

Comment (mversic) [reply client/benches/tps/dev.rs:9]: it's not me who removed them. They have been allowed long ago

Comment (mversic) [reply client/benches/tps/dev.rs:9]: it really doesn't make sense to enable this in every crate. We'll do some lints revision in the future

File: ffi/tests/ffi_import.rs [hunk 4]
```diff
@@ -147,7 +147,7 @@ mod ffi {
         input: SliceRef<FfiTuple2<u32, u32>>,
         output: *mut OutBoxedSlice<FfiTuple2<u32, u32>>,
     ) -> FfiReturn {
-        let input = input.into_rust().map(|slice| slice.to_vec());
+        let input = input.into_rust().map(<[_]>::to_vec);
```
[the remark this answers is not in the document]
Comment (mversic) [reply ffi/tests/ffi_import.rs:150]: forgive me

**Review by Erigara — APPROVED** [review]

**Pull request merged**